code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -1075,49 +1075,105 @@
 		return
 
-	if(plasma_stored >= xeno_caste.plasma_max)
-		to_chat(src, "<span class='notice'>Your [energy] reserves are already at full capacity and can't hold any more.</span>")
-		return
-
 	if(target.stat != DEAD)
-		to_chat(src, "<span class='warning'>You can't steal [energy] from living sisters, ask for some to a drone or a hivelord instead!</span>")
+		to_chat(src, "<span class='warning'>You can't steal [energy] from living sisters!</span>")
 		return
 
-	if(get_dist(src, target) > max_range)
+	if(!Adjacent(target))
 		to_chat(src, "<span class='warning'>You need to be closer to [target].</span>")
 		return
 
-	if(!(target.plasma_stored))
-		to_chat(src, "<span class='notice'>[target] doesn't have any [energy] left to salvage.</span>")
+	if(stagger)
+		to_chat(src, "<span class='xenowarning'>You try to salvage [target]'s essence but are staggered!</span>")
 		return
 
-	to_chat(src, "<span class='notice'>You start salvaging [energy] from [target].</span>")
+	to_chat(src, "<span class='notice'>You start salvaging [target]'s essence...</span>")
 
-	while(target.plasma_stored && plasma_stored >= xeno_caste.plasma_max)
-		if(!do_after(src, salvage_delay, TRUE, 5, BUSY_ICON_HOSTILE) || !check_state())
-			break
+	face_atom(target)
+	animation_attack_on(target)
+	playsound(target, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
 
-		if(!isturf(loc))
-			to_chat(src, "<span class='warning'>You can't absorb [energy] from here!</span>")
-			break
+	if(!do_after(src, salvage_delay, TRUE, 5, BUSY_ICON_HOSTILE, null, TRUE))
+		return
 
-		if(get_dist(src, target) > max_range)
-			to_chat(src, "<span class='warning'>You need to be closer to [target].</span>")
-			break
+	if(!isturf(loc))
+		to_chat(src, "<span class='warning'>You can't absorb [energy] from here!</span>")
+		return
 
-		if(stagger)
-			to_chat(src, "<span class='xenowarning'>Your muscles fail to respond as you try to shake up the shock!</span>")
-			break
+	if(stagger)
+		to_chat(src, "<span class='xenowarning'>You try to salvage [target]'s essence but are staggered!</span>")
+		return
+
+	face_atom(target)
+	animation_attack_on(target)
+	playsound(target, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
+
+	round_statistics.drone_salvage_essence++
+
+	var/absorbed_plasma = round(target.plasma_stored * DRONE_SALVAGE_PLASMA_MULTIPLIER)
+	var/absorbed_evolution = round(target.evolution_stored * DRONE_SALVAGE_EVOLUTION_MULTIPLIER)
+	var/absorbed_upgrade = round(target.upgrade_stored * DRONE_SALVAGE_UPGRADE_MULTIPLIER)
+	var/excess_plasma
+
+	if(absorbed_plasma > (xeno_caste.plasma_max - plasma_stored) ) //Amount exceeding remaining plasma capacity absorbed as health
+		excess_plasma = absorbed_plasma - (xeno_caste.plasma_max - plasma_stored)
+		adjustFireLoss( CLAMP(-excess_plasma, -getFireLoss(), 0 ) )
+		adjustBruteLoss( CLAMP(-excess_plasma, -getBruteLoss(), 0 ) )
```

Comment:
`adjustBruteLoss(-excess_plasma)`

As said, the `CLAMP` is not necessary.

Reply:
All requests have been handled